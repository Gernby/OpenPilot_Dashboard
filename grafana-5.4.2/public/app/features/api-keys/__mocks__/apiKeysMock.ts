import { ApiKey, OrgRole } from 'app/types';

export const getMultipleMockKeys = (numberOfKeys: number): ApiKey[] => {
  const keys: ApiKey[] = [];
  for (let i = 1; i <= numberOfKeys; i++) {
    keys.push({
      id: i,
      name: `test-${i}`,
      role: OrgRole.Viewer,
    });
  }

  return keys;
};

export const getMockKey = (): ApiKey => {
  return {
    id: 1,
    name: 'test',
    role: OrgRole.Admin,
  };
};
