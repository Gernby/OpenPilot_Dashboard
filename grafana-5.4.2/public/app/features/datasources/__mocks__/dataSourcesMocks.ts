import { DataSource } from 'app/types';

export const getMockDataSources = (amount: number): DataSource[] => {
  const dataSources = [];

  for (let i = 0; i <= amount; i++) {
    dataSources.push({
      access: '',
      basicAuth: false,
      database: `database-${i}`,
      id: i,
      isDefault: false,
      jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
      name: `dataSource-${i}`,
      orgId: 1,
      password: '',
      readOnly: false,
      type: 'cloudwatch',
      typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
      url: '',
      user: '',
    });
  }

  return dataSources;
};

export const getMockDataSource = (): DataSource => {
  return {
    access: '',
    basicAuth: false,
    database: '',
    id: 13,
    isDefault: false,
    jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
    name: 'gdev-cloudwatch',
    orgId: 1,
    password: '',
    readOnly: false,
    type: 'cloudwatch',
    typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
    url: '',
    user: '',
  };
};
