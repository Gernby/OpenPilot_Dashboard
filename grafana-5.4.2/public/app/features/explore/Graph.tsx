import $ from 'jquery';
import React, { PureComponent } from 'react';
import moment from 'moment';
import { withSize } from 'react-sizeme';

import 'vendor/flot/jquery.flot';
import 'vendor/flot/jquery.flot.time';
import 'vendor/flot/jquery.flot.selection';
import 'vendor/flot/jquery.flot.stack';

import { RawTimeRange } from 'app/types/series';
import * as dateMath from 'app/core/utils/datemath';
import TimeSeries from 'app/core/time_series2';

import Legend from './Legend';

const MAX_NUMBER_OF_TIME_SERIES = 20;

// Copied from graph.ts
function time_format(ticks, min, max) {
  if (min && max && ticks) {
    const range = max - min;
    const secPerTick = range / ticks / 1000;
    const oneDay = 86400000;
    const oneYear = 31536000000;

    if (secPerTick <= 45) {
      return '%H:%M:%S';
    }
    if (secPerTick <= 7200 || range <= oneDay) {
      return '%H:%M';
    }
    if (secPerTick <= 80000) {
      return '%m/%d %H:%M';
    }
    if (secPerTick <= 2419200 || range <= oneYear) {
      return '%m/%d';
    }
    return '%Y-%m';
  }

  return '%H:%M';
}

const FLOT_OPTIONS = {
  legend: {
    show: false,
  },
  series: {
    lines: {
      linewidth: 1,
      zero: false,
    },
    shadowSize: 0,
  },
  grid: {
    minBorderMargin: 0,
    markings: [],
    backgroundColor: null,
    borderWidth: 0,
    // hoverable: true,
    clickable: true,
    color: '#a1a1a1',
    margin: { left: 0, right: 0 },
    labelMarginX: 0,
  },
  selection: {
    mode: 'x',
    color: '#666',
  },
  // crosshair: {
  //   mode: 'x',
  // },
};

interface GraphProps {
  data: any[];
  height?: string; // e.g., '200px'
  id?: string;
  range: RawTimeRange;
  split?: boolean;
  size?: { width: number; height: number };
  userOptions?: any;
  onChangeTime?: (range: RawTimeRange) => void;
}

interface GraphState {
  showAllTimeSeries: boolean;
}

export class Graph extends PureComponent<GraphProps, GraphState> {
  $el: any;

  state = {
    showAllTimeSeries: false,
  };

  getGraphData() {
    const { data } = this.props;

    return this.state.showAllTimeSeries ? data : data.slice(0, MAX_NUMBER_OF_TIME_SERIES);
  }

  componentDidMount() {
    this.draw();
    this.$el = $(`#${this.props.id}`);
    this.$el.bind('plotselected', this.onPlotSelected);
  }

  componentDidUpdate(prevProps: GraphProps) {
    if (
      prevProps.data !== this.props.data ||
      prevProps.range !== this.props.range ||
      prevProps.split !== this.props.split ||
      prevProps.height !== this.props.height ||
      (prevProps.size && prevProps.size.width !== this.props.size.width)
    ) {
      this.draw();
    }
  }

  componentWillUnmount() {
    this.$el.unbind('plotselected', this.onPlotSelected);
  }

  onPlotSelected = (event, ranges) => {
    if (this.props.onChangeTime) {
      const range = {
        from: moment(ranges.xaxis.from),
        to: moment(ranges.xaxis.to),
      };
      this.props.onChangeTime(range);
    }
  };

  onShowAllTimeSeries = () => {
    this.setState(
      {
        showAllTimeSeries: true,
      },
      this.draw
    );
  };

  draw() {
    const { range, size, userOptions = {} } = this.props;
    const data = this.getGraphData();

    const $el = $(`#${this.props.id}`);
    let series = [{ data: [[0, 0]] }];

    if (data && data.length > 0) {
      series = data.map((ts: TimeSeries) => ({
        color: ts.color,
        label: ts.label,
        data: ts.getFlotPairs('null'),
      }));
    }

    const ticks = (size.width || 0) / 100;
    let { from, to } = range;
    if (!moment.isMoment(from)) {
      from = dateMath.parse(from, false);
    }
    if (!moment.isMoment(to)) {
      to = dateMath.parse(to, true);
    }
    const min = from.valueOf();
    const max = to.valueOf();
    const dynamicOptions = {
      xaxis: {
        mode: 'time',
        min: min,
        max: max,
        label: 'Datetime',
        ticks: ticks,
        timezone: 'browser',
        timeformat: time_format(ticks, min, max),
      },
    };
    const options = {
      ...FLOT_OPTIONS,
      ...dynamicOptions,
      ...userOptions,
    };
    $.plot($el, series, options);
  }

  render() {
    const { height = '100px', id = 'graph' } = this.props;
    const data = this.getGraphData();

    return (
      <>
        {this.props.data &&
          this.props.data.length > MAX_NUMBER_OF_TIME_SERIES &&
          !this.state.showAllTimeSeries && (
            <div className="time-series-disclaimer">
              <i className="fa fa-fw fa-warning disclaimer-icon" />
              {`Showing only ${MAX_NUMBER_OF_TIME_SERIES} time series. `}
              <span className="show-all-time-series" onClick={this.onShowAllTimeSeries}>{`Show all ${
                this.props.data.length
              }`}</span>
            </div>
          )}
        <div id={id} className="explore-graph" style={{ height }} />
        <Legend data={data} />
      </>
    );
  }
}

export default withSize()(Graph);
