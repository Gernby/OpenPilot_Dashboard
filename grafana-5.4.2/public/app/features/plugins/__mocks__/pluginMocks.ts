import { Plugin } from 'app/types';

export const getMockPlugins = (amount: number): Plugin[] => {
  const plugins = [];

  for (let i = 0; i <= amount; i++) {
    plugins.push({
      defaultNavUrl: 'some/url',
      enabled: false,
      hasUpdate: false,
      id: `${i}`,
      info: {
        author: {
          name: 'Grafana Labs',
          url: 'url/to/GrafanaLabs',
        },
        description: 'pretty decent plugin',
        links: ['one link'],
        logos: { small: 'small/logo', large: 'large/logo' },
        screenshots: `screenshot/${i}`,
        updated: '2018-09-26',
        version: '1',
      },
      latestVersion: `1.${i}`,
      name: `pretty cool plugin-${i}`,
      pinned: false,
      state: '',
      type: '',
    });
  }

  return plugins;
};

export const getMockPlugin = () => {
  return {
    defaultNavUrl: 'some/url',
    enabled: false,
    hasUpdate: false,
    id: '1',
    info: {
      author: {
        name: 'Grafana Labs',
        url: 'url/to/GrafanaLabs',
      },
      description: 'pretty decent plugin',
      links: ['one link'],
      logos: { small: 'small/logo', large: 'large/logo' },
      screenshots: 'screenshot/1',
      updated: '2018-09-26',
      version: '1',
    },
    latestVersion: '1',
    name: 'pretty cool plugin 1',
    pinned: false,
    state: '',
    type: '',
  };
};
