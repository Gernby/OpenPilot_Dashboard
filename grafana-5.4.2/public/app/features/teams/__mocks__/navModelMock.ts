export const getMockNavModel = (pageName: string) => {
  return {
    node: {
      active: false,
      icon: 'gicon gicon-team',
      id: `team-${pageName}-2`,
      text: `${pageName}`,
      url: 'org/teams/edit/2/members',
      parentItem: {
        img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
        id: 'team-2',
        subTitle: 'Manage members & settings',
        url: '',
        text: 'test1',
        breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
        children: [
          {
            active: false,
            icon: 'gicon gicon-team',
            id: 'team-members-2',
            text: 'Members',
            url: 'org/teams/edit/2/members',
          },
          {
            active: false,
            icon: 'fa fa-fw fa-sliders',
            id: 'team-settings-2',
            text: 'Settings',
            url: 'org/teams/edit/2/settings',
          },
        ],
      },
    },
    main: {
      img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
      id: 'team-2',
      subTitle: 'Manage members & settings',
      url: '',
      text: 'test1',
      breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
      children: [
        {
          active: true,
          icon: 'gicon gicon-team',
          id: 'team-members-2',
          text: 'Members',
          url: 'org/teams/edit/2/members',
        },
        {
          active: false,
          icon: 'fa fa-fw fa-sliders',
          id: 'team-settings-2',
          text: 'Settings',
          url: 'org/teams/edit/2/settings',
        },
      ],
    },
  };
};
