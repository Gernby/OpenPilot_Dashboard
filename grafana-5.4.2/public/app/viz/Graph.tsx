// Libraries
import $ from 'jquery';
import React, { PureComponent } from 'react';
import { withSize } from 'react-sizeme';
import 'vendor/flot/jquery.flot';
import 'vendor/flot/jquery.flot.time';

// Types
import { TimeRange, TimeSeriesVMs } from 'app/types';

interface GraphProps {
  timeSeries: TimeSeriesVMs;
  timeRange: TimeRange;
  showLines?: boolean;
  showPoints?: boolean;
  showBars?: boolean;
  size?: { width: number; height: number };
}

export class Graph extends PureComponent<GraphProps> {
  static defaultProps = {
    showLines: true,
    showPoints: false,
    showBars: false,
  };

  element: any;

  componentDidUpdate(prevProps: GraphProps) {
    if (
      prevProps.timeSeries !== this.props.timeSeries ||
      prevProps.timeRange !== this.props.timeRange ||
      prevProps.size !== this.props.size
    ) {
      this.draw();
    }
  }

  componentDidMount() {
    this.draw();
  }

  draw() {
    const { size, timeSeries, timeRange, showLines, showBars, showPoints } = this.props;

    if (!size) {
      return;
    }

    const ticks = (size.width || 0) / 100;
    const min = timeRange.from.valueOf();
    const max = timeRange.to.valueOf();

    const flotOptions = {
      legend: {
        show: false,
      },
      series: {
        lines: {
          show: showLines,
          linewidth: 1,
          zero: false,
        },
        points: {
          show: showPoints,
          fill: 1,
          fillColor: false,
          radius: 2,
        },
        bars: {
          show: showBars,
          fill: 1,
          barWidth: 1,
          zero: false,
          lineWidth: 0,
        },
        shadowSize: 0,
      },
      xaxis: {
        mode: 'time',
        min: min,
        max: max,
        label: 'Datetime',
        ticks: ticks,
        timeformat: time_format(ticks, min, max),
      },
      grid: {
        minBorderMargin: 0,
        markings: [],
        backgroundColor: null,
        borderWidth: 0,
        // hoverable: true,
        clickable: true,
        color: '#a1a1a1',
        margin: { left: 0, right: 0 },
        labelMarginX: 0,
      },
    };

    try {
      $.plot(this.element, timeSeries, flotOptions);
    } catch (err) {
      console.log('Graph rendering error', err, flotOptions, timeSeries);
    }
  }

  render() {
    return (
      <div className="graph-panel">
        <div className="graph-panel__chart" ref={e => (this.element = e)} />
      </div>
    );
  }
}

// Copied from graph.ts
function time_format(ticks, min, max) {
  if (min && max && ticks) {
    const range = max - min;
    const secPerTick = range / ticks / 1000;
    const oneDay = 86400000;
    const oneYear = 31536000000;

    if (secPerTick <= 45) {
      return '%H:%M:%S';
    }
    if (secPerTick <= 7200 || range <= oneDay) {
      return '%H:%M';
    }
    if (secPerTick <= 80000) {
      return '%m/%d %H:%M';
    }
    if (secPerTick <= 2419200 || range <= oneYear) {
      return '%m/%d';
    }
    return '%Y-%m';
  }

  return '%H:%M';
}

export default withSize()(Graph);
