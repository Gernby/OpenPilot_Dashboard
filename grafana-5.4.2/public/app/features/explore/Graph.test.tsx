import React from 'react';
import { shallow } from 'enzyme';
import { Graph } from './Graph';
import { mockData } from './__mocks__/mockData';

const setup = (propOverrides?: object) => {
  const props = {
    data: mockData().slice(0, 19),
    range: { from: 'now-6h', to: 'now' },
    ...propOverrides,
  };

  // Enzyme.shallow did not work well with jquery.flop. Mocking the draw function.
  Graph.prototype.draw = jest.fn();

  const wrapper = shallow(<Graph {...props} />);
  const instance = wrapper.instance() as Graph;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disclaimer', () => {
    const { wrapper } = setup({
      data: mockData(),
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show query return no time series', () => {
    const { wrapper } = setup({
      data: [],
    });

    expect(wrapper).toMatchSnapshot();
  });
});
