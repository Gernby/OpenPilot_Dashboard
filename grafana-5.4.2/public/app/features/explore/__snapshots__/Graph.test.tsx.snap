// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<Fragment>
  <div
    className="explore-graph"
    id="graph"
    style={
      Object {
        "height": "100px",
      }
    }
  />
  <Legend
    data={
      Array [
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "+Inf",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.1",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.2",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.4",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "1",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "120",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "20",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "3",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "60",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "8",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "+Inf",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.1",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.4",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "1",
          },
          "values": Array [
            Array [
              1537847900,
              "953",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "120",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "20",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "3",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "60",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "8",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
      ]
    }
  />
</Fragment>
`;

exports[`Render should render component with disclaimer 1`] = `
<Fragment>
  <div
    className="time-series-disclaimer"
  >
    <i
      className="fa fa-fw fa-warning disclaimer-icon"
    />
    Showing only 20 time series. 
    <span
      className="show-all-time-series"
      onClick={[Function]}
    >
      Show all 27
    </span>
  </div>
  <div
    className="explore-graph"
    id="graph"
    style={
      Object {
        "height": "100px",
      }
    }
  />
  <Legend
    data={
      Array [
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "+Inf",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.1",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.2",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.4",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "1",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "120",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "20",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "3",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "60",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/label/:name/values",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "8",
          },
          "values": Array [
            Array [
              1537858100,
              "16",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "+Inf",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.1",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "0.4",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "1",
          },
          "values": Array [
            Array [
              1537847900,
              "953",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "120",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "20",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "3",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "60",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/metrics",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "8",
          },
          "values": Array [
            Array [
              1537858060,
              "1195",
            ],
            Array [
              1537858080,
              "1195",
            ],
            Array [
              1537858100,
              "1195",
            ],
          ],
        },
        Object {
          "metric": Object {
            "__name__": "prometheus_http_request_duration_seconds_bucket",
            "handler": "/query",
            "instance": "localhost:9090",
            "job": "prometheus",
            "le": "+Inf",
          },
          "values": Array [
            Array [
              1537858100,
              "55",
            ],
            Array [
              1537861960,
              "1",
            ],
            Array [
              1537861980,
              "1",
            ],
          ],
        },
      ]
    }
  />
</Fragment>
`;

exports[`Render should show query return no time series 1`] = `
<Fragment>
  <div
    className="explore-graph"
    id="graph"
    style={
      Object {
        "height": "100px",
      }
    }
  />
  <Legend
    data={Array []}
  />
</Fragment>
`;
